### ISSUE TITLE
[4.0] Errors when saving article with long content

### ISSUE THREAD

**Issue by regularlabs** [joomla/joomla-cms#29873]
There seems to be an issue when saving content of more than 65,535 characters.
65,535 is the length of a `TEXT` field, even though the `introtext` and `fulltext` fields are set to `MEDIUMTEXT` in the database.

The error is:
```
Error
Save failed with the following error: Data too long for column 'description' at row 1
```

Once that error has happened, stuff gets broken in the database and it is no longer possible to correct the issue by deleting content. When you try to save again with a shorter text, it will throw this error:
```
Error
Save failed with the following error: Incorrect integer value: '' for column 'hits' at row 1
```

It took some digging, but the issue is down to the **Finder** plugin.
This gets triggered on `onAfterSaveContent`. And tries to save the full content to the `description` field in the `finder_links` table. That field has the type `TEXT`. So it chokes.

So not only is there an issue with that field length, there is also an issue with error handling.
It shouldn't be allowed that if a plugin has an issue with saving meta data, it causes the initial content (article) to break and go haywire.

**Comment by wilsonge** [comment]
Presumably this is a J3 issue as well?

**Comment by ReLater** [comment]
> Presumably this is a J3 issue as well?

J3 saves the article and the description in finder_links but is cropping the description silently. Then I get an

```
Error
Save failed with the following error: The table '#__finder_tokens' is full
```
which is another issue I think. It's at least misleading because the article is saved.

J4 saves neither the article nor the description.

**Comment by Hackwar** [comment]
This should be fixed with #30008
